### PR TITLE
Dockerfile lint 3

### PR DESCRIPTION
**Pull request by ammarlakis**
**Description**

Changes proposed in this pull request:

- Fixing linting errors in Dockerfiles. Patch 3 includes the components:
  - components/iam-kubeconfig-service/Dockerfile
  - components/kyma-operator/Dockerfile
  - components/service-binding-usage-controller/Dockerfile
  - components/uaa-activator/Dockerfile
  - tests/integration/api-gateway/Dockerfile
  - tests/integration/apiserver-proxy/Dockerfile
  - tests/integration/dex/Dockerfile
  - tests/rafter/Dockerfile
  - tests/service-catalog/Dockerfile
  - tools/event-subscriber/Dockerfile
  - tools/gitserver/Dockerfile

**Related issue(s)**
Related to #12854

### CODE REVIEW

**Review by mfaizanse — APPROVED** [review]

**Review by dbadura — CHANGES_REQUESTED** [review]
Please bump git server: https://github.com/kyma-project/kyma/blob/main/resources/serverless/values.yaml#L117

**Review by werdes72 — DISMISSED** [review]

File: tools/gitserver/Dockerfile [hunk 1]
```diff
@@ -1,7 +1,6 @@
 FROM alpine:3.12
 
-RUN apk update &&\
```
Comment (k15r) [inline tools/gitserver/Dockerfile:3]: is there a reason to not update the package list?

File: tools/event-subscriber/Dockerfile [hunk 2]
```diff
@@ -17,6 +17,8 @@ RUN CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -o event-subscriber
 FROM scratch
 LABEL source=git@github.com:kyma-project/kyma.git
 
+WORKDIR /app
```
Comment (k15r) [inline tools/event-subscriber/Dockerfile:20]: ?? why do we change the workdir to /app and not change it in the `CMD` in line 27? imho this introduces an error

**Review by k15r — CHANGES_REQUESTED** [review]

File: tools/gitserver/Dockerfile (hunk 1 above)
Comment (ammarlakis) [reply tools/gitserver/Dockerfile:3]: we are updating the packages with `--no-cache` flag, just not storing it.

File: tests/service-catalog/Dockerfile [hunk 3]
```diff
@@ -15,11 +15,13 @@ RUN CGO_ENABLED=0 go test -tags=acceptance -c ./test -o /servicecatalog.test
 FROM alpine:3.10
 RUN apk add --no-cache curl
 
+WORKDIR /app
```
Comment (wozniakjan) [inline tests/service-catalog/Dockerfile:18]: this influences the following `COPY` and `ENTRYPOINT` arguments, please make sure they are unified about the workdir.

File: tests/service-catalog/Dockerfile [hunk 4]
```diff
@@ -15,11 +15,13 @@ RUN CGO_ENABLED=0 go test -tags=acceptance -c ./test -o /servicecatalog.test
...
+
 COPY --from=builder env-tester.bin /go/bin/env-tester.bin
 COPY --from=builder ./servicecatalog.test .
 COPY ./licenses/ /app/licenses
 COPY --from=builder /entrypoint.sh .
 
 LABEL source=git@github.com:kyma-project/kyma.git
 
-ENTRYPOINT ./entrypoint.sh
+ENTRYPOINT ["./entrypoint.sh"]
```
Comment (wozniakjan) [inline tests/service-catalog/Dockerfile:27]: I guess at least this should be now `["/entrypoint.sh"]`, because of the `WORKDIR /app` on line 18.

Comment (ammarlakis) [reply tests/service-catalog/Dockerfile:27]: it's already copied to the local directory. See line 23

Comment (wozniakjan) [reply tests/service-catalog/Dockerfile:27]: ah, my bad

**Review by wozniakjan — DISMISSED** [review]
/lgtm
/hold
change to files owned by team gophers look good, feel free to unhold when others are satisfied as well

**Review by werdes72 — DISMISSED** [review]

**Review by dbadura — APPROVED** [review]

**Review by mfaizanse — APPROVED** [review]

**Review by mfaizanse — APPROVED** [review]

**Review by k15r — APPROVED** [review]

**Comment by ammarlakis** [comment]
Will be split into smaller PRs.

**Pull request closed without merging**